src/server/process_request.c

```diff
@@ -316,17 +317,24 @@ req_register_sched(conn_t *conn, struct batch_request *preq)
 		goto rerr;
 	}
 
-	if (sched->sc_conn_addr != conn->cn_addr) {
-		if (pbs_conf.pbs_primary != NULL && pbs_conf.pbs_secondary != NULL) {
-			pbs_net_t addr = get_hostaddr(pbs_conf.pbs_primary);
+	if (pbs_conf.pbs_primary != NULL && pbs_conf.pbs_secondary != NULL) {
+		are_primary = are_we_primary();
+		pbs_net_t addr;
+		if (are_primary == FAILOVER_PRIMARY) {
+			addr = get_hostaddr(pbs_conf.pbs_primary);
 			if (addr != conn->cn_addr) {
-				addr = get_hostaddr(pbs_conf.pbs_secondary);
-				if (addr != conn->cn_addr) {
-					rc = PBSE_BADHOST;
-					goto rerr;
-				}
+				rc = PBSE_BADHOST;
+				goto rerr;
 			}
-		} else {
+		}
+		else if (are_primary == FAILOVER_SECONDARY) {
+			addr = get_hostaddr(pbs_conf.pbs_secondary);
+			if (addr != conn->cn_addr) {
+				rc = PBSE_BADHOST;
+				goto rerr;
+			}
+		}
+		else {
```

Comment:
shall we have the closing braces, else and the opening braces on the same line.

Reply:
I run clang format on the file, and the result was closing brace, else (and open brace) on the same line. I made the necessary changes in a new commit.